src/sentry/uptime/grouptype.py

```diff
@@ -1,15 +1,289 @@
+
+        issue_creation_allowed = (
+            detector_issue_creation_enabled
+            and issue_creation_flag_enabled
+            and host_provider_enabled
+        )
+
+        # XXX(epurkhiser): We currently are duplicating the detector state onto
+        # the uptime_subscription when the detector changes state. Once we stop
+        # using this field we can drop this update logic.
+        #
+        # We ONLY do this when detector issue creation is enabled, otherwise we
+        # let the legacy uptime consumer handle this.
+        if detector_issue_creation_enabled:
+            if evaluation.priority == DetectorPriorityLevel.OK:
+                uptime_status = UptimeStatus.OK
+            elif evaluation.priority != DetectorPriorityLevel.OK:
+                uptime_status = UptimeStatus.FAILED
+
+            uptime_subscription.update(
+                uptime_status=uptime_status,
+                uptime_status_update_date=django_timezone.now(),
+            )
+
+        if not host_provider_enabled:
+            metrics.incr(
+                "uptime.result_processor.restricted_by_provider",
+                sample_rate=1.0,
+                tags={
+                    "host_provider_id": host_provider_id,
+                    **metric_tags,
+                },
+            )
+
+        result_creates_issue = isinstance(evaluation.result, IssueOccurrence)
+        result_resolves_issue = isinstance(evaluation.result, StatusChangeMessage)
+
+        if result_creates_issue:
+            metrics.incr(
+                "uptime.detector.will_create_issue",
+                tags=metric_tags,
+                sample_rate=1.0,
+            )
+            # XXX(epurkhiser): This logging includes the same extra arguments
+            # as the `uptime_active_sent_occurrence` log in the consumer for
+            # legacy creation
+            logger.info(
+                "uptime.detector.will_create_issue",
+                extra={
+                    "project_id": self.detector.project_id,
+                    "url": uptime_subscription.url,
+                    **data_packet.packet.check_result,
+                },
+            )
+        if result_resolves_issue:
+            metrics.incr(
+                "uptime.detector.will_resolve_issue",
+                sample_rate=1.0,
+                tags=metric_tags,
+            )
+            logger.info(
+                "uptime.detector.will_resolve_issue",
+                extra={
+                    "project_id": self.detector.project_id,
+                    "url": uptime_subscription.url,
+                    **data_packet.packet.check_result,
+                },
+            )
+
+        # Reutning an empty dict effectively causes the detector processor to
+        # bail and not produce an issue occurrence.
+        if result_creates_issue and not issue_creation_allowed:
+            return {}
+
+        return result
```

Comment:
curious if y'all have any thoughts about this evaluation bit -- is there anything for you that would make this easier to implement? think it's okay to just invoke the super and handle it like this?

Reply:
For now I think this is fine. Once we remove a lot of the legacy issue creation code for uptime some of this is going to go away and it will be more clear what can be factored and abstracted.